server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala

```diff
@@ -96,12 +117,94 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
+                                             mailboxSession: MailboxSession): Either[IllegalArgumentException, MultipartBuilder] = {
+    val multipartBuilder = MultipartBuilder.create(SubType.MIXED_SUBTYPE)
+    val bodypartBuilder = BodyPartBuilder.create()
+
+    attachments
+      .map(attachment => getAttachmentMetadata(attachment, attachmentManager, mailboxSession))
+      .map(attachmentMetadataList => attachmentMetadataList
+        .flatMap(attachmentAndMetadata => loadAttachment(attachmentAndMetadata._1, attachmentAndMetadata._2, attachmentContentLoader, mailboxSession)))
+      .sequence
+
+    val maybeAttachments: Either[IllegalArgumentException, List[(Attachment, AttachmentMetadata, Array[Byte])]] =
+      attachments
+        .filter(!_.isInline)
+        .map(attachment => getAttachmentMetadata(attachment, attachmentManager, mailboxSession))
+        .map(attachmentMetadataList => attachmentMetadataList
+          .flatMap(attachmentAndMetadata => loadAttachment(attachmentAndMetadata._1, attachmentAndMetadata._2, attachmentContentLoader, mailboxSession)))
+        .sequence
+
+    maybeAttachments.map(list => {
+      list.foldLeft(multipartBuilder) {
+        case (builder, (attachment, attachmentMetadata, content)) =>
+          bodypartBuilder.setBody(content, attachment.`type`.value)
+            .setField(contentTypeField(attachment, attachmentMetadata))
+            .setContentDisposition(attachment.disposition.getOrElse(Disposition.ATTACHMENT).value)
+
+          if (attachment.cid.isDefined) {
+            bodypartBuilder.setField(new RawField("Content-ID", attachment.cid.get))
+          }
+
+          builder.addBodyPart(bodypartBuilder)
+          builder
+      }
     })
+  }
+
+  private def contentTypeField(attachment: Attachment, attachmentMetadata: AttachmentMetadata): ContentTypeField = {
+    val typeAsField: ContentTypeField = attachmentMetadata.getType.asMime4J
+    if (attachment.name.isDefined) {
+      Fields.contentType(typeAsField.getMimeType,
+        Map.newBuilder[String, String]
+          .addAll(parametersWithoutName(typeAsField))
+          .addOne("name", EncoderUtil.encodeEncodedWord(attachment.name.get.value, Usage.TEXT_TOKEN))
+          .result
+          .asJava);
+    }
+    typeAsField;
+  }
+
+  private def parametersWithoutName(typeAsField: ContentTypeField): Map[String, String] = {
+    typeAsField.getParameters
+      .asScala
+      .filter(!_._1.equals("name"))
+      .toMap
+  }
+
+  private def getAttachmentMetadata(attachment: Attachment,
+                                    attachmentManager: AttachmentManager,
+                                    mailboxSession: MailboxSession): Either[AttachmentNotFoundException, (Attachment, AttachmentMetadata)] = {
+    Try(attachmentManager.getAttachment(AttachmentId.from(attachment.blobId.value.toString), mailboxSession))
+      .fold(e => Left(AttachmentNotFoundException(e, s"Attachment not found: ${attachment.blobId.value}")),
```

Comment:
You are considering MailboxException as "the attachment is not here".

I would suggest to switch the method type to `Either[Exception, (Attachment, AttachmentMetadata)]`